coral-service/src/main/java/com/linkedin/coral/coralservice/utils/IncrementalUtils.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.coralservice.utils;
+
+import org.apache.calcite.rel.RelNode;
+
+import com.linkedin.coral.hive.hive2rel.HiveToRelConverter;
+import com.linkedin.coral.incremental.RelNodeIncrementalTransformer;
+import com.linkedin.coral.spark.CoralSpark;
+
+import static com.linkedin.coral.coralservice.utils.CoralProvider.*;
+
+
+public class IncrementalUtils {
+
+  public static String getIncrementalQueryFromUserSql(String query) {
+    RelNode originalNode = new HiveToRelConverter(hiveMetastoreClient).convertSql(query);
+    RelNode incrementalRelNode = RelNodeIncrementalTransformer.convertRelIncremental(originalNode);
+    CoralSpark coralSpark = CoralSpark.create(incrementalRelNode);
+    return coralSpark.getSparkSql();
+  }
```

Comment:
Should this take a dialect as input? Right now it is assumed to be Spark, but what if we want to rewrite a Trino query in the future? This should apply all the way to the callers of this API on the client side.

Reply:
Let us unify the Dialect representation with whatever being used in translation utils. Consider creating an ENUM for dialects?

Reply:
Looks like there is a defined mapping of supported dialects in the translation controller. Thinking of pulling from that for this endpoint.